standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1740,4039 +649,413 @@ public long getLatestTxnIdInConflict(long txnid) throws MetaException {
+        .execute(jdbcResource);
   }
-
+  
   /**
-   * Get txns from the list that are committed.
-   * @param txnIds list of txns to be evaluated for committed state
-   * @param stmt db statement
+   * Catalog hasn't been added to transactional tables yet, so it's passed in but not used.
    */
-  private String getCommittedTxns(List<Long> txnIds, Statement stmt) throws SQLException {
-    List<String> queries = new ArrayList<>();
-    StringBuilder prefix = new StringBuilder();
-
-    // Check if any of the txns in the list are committed.
-    prefix.append("SELECT \"CTC_TXNID\" FROM \"COMPLETED_TXN_COMPONENTS\" WHERE ");
-    TxnUtils.buildQueryWithINClause(conf, queries, prefix, new StringBuilder(),
-        txnIds, "\"CTC_TXNID\"", false, false);
-    StringBuilder txnInfo = new StringBuilder();
-
-    for (String query : queries) {
-      LOG.debug("Going to execute query <{}>", query);
-      try (ResultSet rs = stmt.executeQuery(query)) {
-        while (rs.next()) {
-          long txnId = rs.getLong(1);
-          txnInfo.append("{").append(txnId).append(",c}");
-        }
-      }
-    }
-    return txnInfo.toString();
+  @Override
+  public void onRename(String oldCatName, String oldDbName, String oldTabName, String oldPartName,
+      String newCatName, String newDbName, String newTabName, String newPartName)
+      throws MetaException {
+    new OnRenameFunction(oldCatName, oldDbName, oldTabName, oldPartName, newCatName, newDbName, newTabName, newPartName).execute(jdbcResource);
   }
-
   /**
-   * Used to raise an informative error when the caller expected a txn in a particular TxnStatus
-   * but found it in some other status
+   * For testing only, do not use.
    */
-  private static void raiseTxnUnexpectedState(TxnStatus actualStatus, long txnid)
-    throws NoSuchTxnException, TxnAbortedException {
-    switch (actualStatus) {
-      case ABORTED:
-        throw new TxnAbortedException("Transaction " + JavaUtils.txnIdToString(txnid) + " already aborted");
-      case COMMITTED:
-        throw new NoSuchTxnException("Transaction " + JavaUtils.txnIdToString(txnid) + " is already committed.");
-      case UNKNOWN:
-        throw new NoSuchTxnException("No such transaction " + JavaUtils.txnIdToString(txnid));
-      case OPEN:
-        throw new NoSuchTxnException(JavaUtils.txnIdToString(txnid) + " is " + TxnStatus.OPEN);
-      default:
-        throw new IllegalArgumentException("Unknown TxnStatus " + actualStatus);
-    }
+  @VisibleForTesting
+  public int numLocksInLockTable() {
+    return Objects.requireNonNull(
+        jdbcResource.getJdbcTemplate().queryForObject("SELECT COUNT(*) FROM \"HIVE_LOCKS\"", new MapSqlParameterSource(), Integer.TYPE),
+        "This never should be null, it's just to suppress warnings");
   }
+
   /**
-   * Returns the state of the transaction with {@code txnid} or throws if {@code raiseError} is true.
+   * For testing only, do not use.
```

Comment:
we shouldn't place test stuff in production code

Reply:
I agree, but I think this should be handled separately. Currently this is the only way to alter the `timeout` field

Reply:
ok, could we please add TODO, so we'll this all todo stuff in a report later